codeworld-api/src/CodeWorld/Driver.hs

```diff
@@ -210,7 +210,8 @@ type DrawState = (Double, Double, Double, Double, Double, Double, Maybe Color)
 type NodeId = Int
 
 pictureToDrawing :: Picture -> Drawing
-pictureToDrawing (SolidPolygon _ pts s) = Shape $ polygonDrawer pts s
+pictureToDrawing (SolidClosedCurve _ pts s) = Shape $ polygonDrawer pts True
```

Comment:
but then it gives me errors that I need to remove the Bool fields of the ```solidPolygon```, ```solidClosedCurve``` and ```solidClosedLoop``` functions (lines 138, 171, 175) in Pictures.hs, and once I resolve those, I've been having difficulty trying to get type-checking in Driver.hs because it keeps wanting me to change values in ```PictureToDrawing``` and ```picObj```. :/

Reply:
So getting rid of the `Bool` fields of `solidPolygon`/`solidClosedCurve` and `solidClosedLoop` is an expected outcome of that change.  What is the type error that you run into in `Driver.hs`?

Reply:
I get three. One on line 213 in ```Driver.hs``` expecting two arguments (which is fixable), then one in line 442 wanting two arguments (but not sure I should dick with that), and one on 576 wanting two arguments. I've tried in the past to fix those and ended up with the error I got two days about ```polygonDrawer``` (above image error)

Reply:
So the errors in `picToObj` you should fix in the same way that you fixed `pictureToDrawing`, by supplying the correct `Bool` based on whether you are drawing a `SolidPolygon` or `SolidClosedCurve`

For the error in `getPictureCS`, the correct thing to do is to delete the unused argument

Reply:
Thank you!

Reply:
You're welcome!